cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AddBrokerPlan.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.detector;
+
+import com.linkedin.kafka.cruisecontrol.metricsreporter.exception.UnknownVersionException;
+import java.nio.ByteBuffer;
+import java.util.HashSet;
+import java.util.Set;
+
+
+public class AddBrokerPlan extends MaintenancePlan {
+  public static final byte PLAN_VERSION = 0;
+  private final Set<Integer> _brokers;
+
+  public AddBrokerPlan(long timeMs, int brokerId, Set<Integer> brokers) {
+    super(MaintenanceEventType.ADD_BROKER, timeMs, brokerId);
+    if (brokers == null || brokers.isEmpty()) {
+      throw new IllegalArgumentException("Missing brokers for the plan.");
+    }
+    if (brokers.size() > Short.MAX_VALUE) {
+      throw new IllegalArgumentException(String.format("Cannot add more than %d brokers (attempt: %d).",
+                                                       Short.MAX_VALUE, brokers.size()));
+    }
+    _brokers = brokers;
+  }
+
+  @Override
+  public byte planVersion() {
+    return PLAN_VERSION;
+  }
+
+  public Set<Integer> brokers() {
+    return _brokers;
+  }
+
+  @Override
+  public ByteBuffer toBuffer(int headerSize) {
```

Comment:
Do you want to add a magic number and/or CRC to detect corruption?

Reply:
I believe we already have a `magic number` for each plan (please see `MaintenancePlan#planVersion()`) -- am I missing something?
-- Added `crc` to verify the integrity -- thanks for the suggestion!